### ISSUE TITLE
Encoding::UndefinedConversionError

### ISSUE THREAD

**Issue by bsv9** [rack/rack#732]
Found this errors in my error.log

```
2014-09-08 21:16:58 +0000: Read error: #<Encoding::UndefinedConversionError: "\xC3" to UTF-8 in conversion from ASCII-8BIT to UTF-8 to US-ASCII>
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/rack-1.4.5/lib/rack/commonlogger.rb:46:in `write'
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/rack-1.4.5/lib/rack/commonlogger.rb:46:in `log'
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/puma-2.8.0/lib/puma/rack_patch.rb:20:in `block in call'
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/puma-2.8.0/lib/puma/server.rb:642:in `call'
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/puma-2.8.0/lib/puma/server.rb:642:in `block in handle_request'
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/puma-2.8.0/lib/puma/server.rb:642:in `each'
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/puma-2.8.0/lib/puma/server.rb:642:in `handle_request'
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/puma-2.8.0/lib/puma/server.rb:361:in `process_client'
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/puma-2.8.0/lib/puma/server.rb:254:in `block in run'
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/puma-2.8.0/lib/puma/thread_pool.rb:92:in `call'
/usr/local/rvm/gems/ruby-1.9.3-p484@global/gems/puma-2.8.0/lib/puma/thread_pool.rb:92:in `block in spawn_thread'
```

option stdout_redirect

```
stdout_redirect 'log/access.log', 'log/error.log', true
```

ENV: ruby 1.9.3-p484, puma-2.7.1
OS: Ubuntu 12.10 x86_64

**Comment by jeremyevans** [comment]
So it seems this issue is related to how Rack::CommonLogger's logger object (which is passed as an argument) handles encoded data.  That doesn't seem to be a bug in Rack itself, and I don't think Rack::CommonLogger can fix the issue, because it can only assume the logger object supports `write` or `<<`.  

I think Rack's SPEC's handling of encoding (treating input as binary, since you don't know the encoding) is fine and should not be changed.

Basically, this issue can be fixed by the software using Rack::CommonLogger providing an argument that supports `write` or `<<` with strings in any encoding.  As this isn't a bug in Rack, I'm going to close the issue now.
